src/components/workspace-utils.js

```diff
@@ -340,6 +342,46 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceStarControl = ({ workspace, starredWorkspaceIds, onUpdate, style }) => {
+  const [updating, setUpdating] = useState(false)
+
+  const { workspace: { workspaceId } } = workspace
+  const isStarred = _.includes(workspaceId, starredWorkspaceIds)
+
+  const toggleStar = _.flow(
+    Utils.withBusyState(setUpdating),
+    withErrorReporting(`Unable to ${isStarred ? 'unstar' : 'star'} workspace`)
+  )(async star => {
+    if (star) {
+      if (_.size(starredWorkspaceIds) < 50) {
+        const updatedWorkspaceIds = _.concat(starredWorkspaceIds, [workspaceId])
+        await Ajax().User.profile.setPreferences({ starredWorkspaces: _.join(',', updatedWorkspaceIds) })
+        authStore.update(_.set('profile.starredWorkspaces', updatedWorkspaceIds))
+      } else {
+        reportError('Unable to star workspace.',
+          'A maximum of 50 workspaces can be starred. Please un-star another workspace before starring this workspace.')
+      }
+    } else {
+      const updatedWorkspaceIds = _.without([workspaceId], starredWorkspaceIds)
+      await Ajax().User.profile.setPreferences({ starredWorkspaces: _.join(',', updatedWorkspaceIds) })
+      authStore.update(_.set('profile.starredWorkspaces', updatedWorkspaceIds))
+    }
+    onUpdate(star)
+  })
+
+  return h(Interactive, {
+    as: 'span',
+    role: 'checkbox',
+    'aria-checked': isStarred,
+    tooltip: isStarred ? 'Unstar this workspace' : 'Star this workspace',
```

Comment:
![image](https://user-images.githubusercontent.com/484484/179569318-1efe6827-727d-418b-a1ad-41211cbd36da.png)

Reply:
I switched this from an Interactive to a Clickable, which seems to improve the accessibility. I also added an aria label to announce when a workspace is starred, rather than just "checkbox checked". I am not sure if I should also announce if a workspace is unstarred- I chose not to because that is the default state for a workspace and I didn't want it to be too chatty. The tooltip also works as expected now as a Clickable

Improved in acb0d64fbd2c7f03e7ad3cb83a5547cc60d17968

Also added a better description of what starring a workspace does in https://github.com/DataBiosphere/terra-ui/pull/3229/commits/aa33ae1d07f13e12701536423b2cce5dac68b702